code/mission/missionparse.cpp

```diff
@@ -5840,21 +5843,18 @@ void post_process_mission()
 
 			// entering this if statement will result in program termination!!!!!
 			// print out an error based on the return value from check_sexp_syntax()
+			// G5K: now entering this statement simply aborts the mission load
 			if ( result ) {
 				SCP_string sexp_str;
 				SCP_string error_msg;
 
 				convert_sexp_to_string(sexp_str, i, SEXP_ERROR_CHECK_MODE);
 				truncate_message_lines(sexp_str, 30);
 				sprintf(error_msg, "%s.\n\nIn sexpression: %s\n(Error appears to be: %s)", sexp_error_message(result), sexp_str.c_str(), Sexp_nodes[bad_node].text);
+				Warning(LOCATION, "%s", error_msg.c_str());
 
-				if (!Fred_running) {
-					nprintf(("Error", "%s", error_msg.c_str()));
-					Error(LOCATION, "%s", error_msg.c_str());
-				} else {
-					nprintf(("Warning", "%s", error_msg.c_str()));
-					Warning(LOCATION, "%s", error_msg.c_str());
-				}
+				// syntax errors are unrecoverable, so abort
+				return -3;
```

Comment:
This should use an `enum class` instead of a magic number or possibly a `bool` if there are only two possible return values.

Reply:
It's magic numbers all the way down though.  I pushed another commit but it makes this PR much more complicated.